processing/src/main/java/org/apache/druid/timeline/partition/OvershadowableManager.java

```diff
@@ -65,7 +65,7 @@
  *
  * This class is not thread-safe.
  */
-class OvershadowableManager<T extends Overshadowable<T>>
+class OvershadowableManager<T extends Overshadowable<T>> implements PartitionHolderContents<T>
```

Comment:
Maybe rename this to indicate that it is used only when segment locking is used.